tools/cache.py

```diff
@@ -23,13 +23,20 @@
 cachelock_name = None
 
 
+def is_writable(path):
+  return os.access(path, os.W_OK)
+
+
 def acquire_cache_lock(reason):
   global acquired_count
   if config.FROZEN_CACHE:
     # Raise an exception here rather than exit_with_error since in practice this
     # should never happen
     raise Exception('Attempt to lock the cache but FROZEN_CACHE is set')
 
+  if not is_writable(cachedir):
+    utils.exit_with_error(f'Cache directory "{cachedir}" is not writable while accesing cache for: {reason}')
```

Comment:
```suggestion
    utils.exit_with_error(f'Cache directory "{cachedir}" is not writable while accessing cache for: {reason}')
```

Reply:
Also, maybe we can link to a doc page on the website (which explains `CACHE` etc.)?

Reply:
I don't know if we have such a page that describes each of the config settings like that.  Perhaps we should.   The best I could find is https://emscripten.org/docs/building_from_source/configuring_emscripten_settings.html. 

Remember that all this stuff only applies to the more advanced users out there, not users of emsdk.

Reply:
Agreed it's more advanced users, but I'm just trying to make this breaking change less annoying for them.

I see we have this page

https://emscripten.org/docs/tools_reference/emcc.html

whose entry for `--cache` (I can't seem to link to it directly) mentions `CACHE`. Maybe that's a good place to link to?

Reply:
Ok done.